### DISCUSSION TITLE
Advanced timer don't have a method to set RCR register

### DISCUSSION THREAD

**Discussion by victorandrehc** [modm-io/modm#979] [Ideas]
The RCR register is very useful to set a fix number of pulses on advanced timers when is PWM mode. WIth such register one can configure a PWM signal with a given dutycycle and a fixed amout of pulses in the pulsetrain. This can be used for applications like:

- Controlling a stepper motor without any CPU management once the number of steps is decided
- Using it as an ADC trigger, that is linked to a dma buffer and a certain amount of samples has to be done in sequence.

I could also gladly create a pullrequest with those changes.

**Comment by salkinium** [comment]
Yes, go for it!

**Reply by salkinium** [reply]
LOL, there was an attempt:
https://github.com/modm-io/modm/blob/develop/src/modm/platform/timer/stm32/advanced_base.hpp.in#L42
